onnxruntime/test/testdata/custom_op_library/custom_op_library.cc

```diff
@@ -144,25 +103,36 @@ struct CustomOpTwo : Ort::CustomOpBase<CustomOpTwo, KernelTwo> {
   size_t GetOutputTypeCount() const { return 1; };
   ONNXTensorElementDataType GetOutputType(size_t /*index*/) const { return ONNX_TENSOR_ELEMENT_DATA_TYPE_INT32; };
 
-} c_CustomOpTwo;
-
-OrtStatus* ORT_API_CALL RegisterCustomOps(OrtSessionOptions* options, const OrtApiBase* api) {
-  OrtCustomOpDomain* domain = nullptr;
-  const OrtApi* ortApi = api->GetApi(ORT_API_VERSION);
-
-  if (auto status = ortApi->CreateCustomOpDomain(c_OpDomain, &domain)) {
-    return status;
-  }
-
-  AddOrtCustomOpDomainToContainer(domain, ortApi);
+};
 
-  if (auto status = ortApi->CustomOpDomain_Add(domain, &c_CustomOpOne)) {
-    return status;
-  }
+static void AddOrtCustomOpDomainToContainer(Ort::CustomOpDomain&& domain) {
+  static std::vector<Ort::CustomOpDomain> ort_custom_op_domain_container;
+  static std::mutex ort_custom_op_domain_mutex;
+  std::lock_guard<std::mutex> lock(ort_custom_op_domain_mutex);
+  ort_custom_op_domain_container.push_back(std::move(domain));
+}
 
-  if (auto status = ortApi->CustomOpDomain_Add(domain, &c_CustomOpTwo)) {
-    return status;
+OrtStatus* ORT_API_CALL RegisterCustomOps(OrtSessionOptions* options, const OrtApiBase* api) {
+  Ort::Global<void>::api_ = api->GetApi(ORT_API_VERSION);
```

Comment:
I don't think custom ops should have to explicitly touch internal globals from the C++ api header in order to utilize those APIs. Is your new `InitProviderOrtApi()` meant to provide a way to initialize the global? If so, I'm not exactly sure how this would be accessed by a custom op dll. Can you please elaborate?

Or is the goal to address this in a future PR?

Reply:
I agree. This is for providers. Need to come up with another way for CustomOp DLLs.